tobac/utils/general.py

```diff
@@ -631,6 +631,224 @@ def combine_feature_dataframes(
+            # initiate new column in feature dataframe if it does not already exist
+            if stats_name not in features.columns:
+                features[stats_name] = None
+                # if function is given as a tuple, take the input parameters provided
+            if type(func_dict[stats_name]) is tuple:
+                func = func_dict[stats_name][0]
+                # check that key word arguments are provided as dictionary
+                if not type(func_dict[stats_name][1]) is dict:
+                    raise TypeError(
+                        "Tuple must contain dictionary with key word arguments for function."
+                    )
+                else:
+                    kwargs = func_dict[stats_name][1]
+                    # default needs to be sequence when function output is array-like
+                    output = func(np.random.rand(1, 10), **kwargs)
+                    if hasattr(output, "__len__"):
+                        default = np.full(output.shape, default)
+                    stats = np.array(
+                        [
+                            func(
+                                *[
+                                    field.ravel()[argsorted[bins[i - 1] : bins[i]]]
+                                    for field in fields
+                                ],
+                                **kwargs,
+                            )
+                            if bins[i] > bins[i - 1]
+                            else default
+                            for i in index
+                        ]
+                    )
+            # otherwise apply function on region without any input parameter
+            else:
+                func = func_dict[stats_name]
+                # default needs to be sequence when function output is array-like
+                output = func(np.random.rand(1, 10))
+                if hasattr(output, "__len__"):
+                    default = np.full(output.shape, default)
+
+                stats = np.array(
+                    [
+                        func(
+                            *[
+                                field.ravel()[argsorted[bins[i - 1] : bins[i]]]
+                                for field in fields
+                            ]
+                        )
+                        if bins[i] > bins[i - 1]
+                        else default
+                        for i in index
+                    ]
+                )
+
+            # add results of computed statistics to feature dataframe with column name given per func_dict
+            for idx, label in enumerate(np.unique(labels[labels > 0])):
+
+                # test if values are scalars
+                if not hasattr(stats[idx], "__len__"):
+                    # if yes, we can just assign the value to the new column and row of the respective feature
+                    features.loc[features[id_column] == label, stats_name] = stats[idx]
```

Comment:
When assigning the results to the dataframe, we loop over only values in the segment labels, but the stats array has elements for every contiguous integer between 1 and the highest label. If the segmentation is not contiguous (which is often the case) we end up assigning the wrong values to the results.

We can either fix this by assigning based on the label value, or limiting index so that we only calculate stats for labels that exist in the df. I'll look at implementing these and see if it fixes things

Reply:
Good catch, I totally missed that! I think it would be more elegant to define the index so that we only calculate statistics for labels that exist in the dataframe but I am fine with whatever is easier to implement. Thank you for your help with this!

Reply:
Have just pushed a quick fix for this

Reply:
Happy with this fix, thank you!